src/gui/dialog.hpp

```diff
@@ -81,6 +82,24 @@ class Dialog
     MenuManager::instance().set_dialog(std::move(dialog));
   }
 
+  // the callback is called regardless the state(enabled/disabled) of confirmation dialog
+  static void show_confirmation(const std::string& text, const std::function<void ()>& callback)
+  {
+    if (g_config->confirmation_dialog)
```

Comment:
This check should go in game_menu.cpp

Reply:
It'll create duplicate 'if' just like before.